test-case/latency-baseline.sh

```diff
@@ -0,0 +1,228 @@
+    printf("\"min_frames\":%0.3f, \"max_frames\":%0.3f, ", min(latency_frames), max(latency_frames))
+    printf("\"avg_ms\":%0.3f, ", (length(latency_ms) ? sum(latency_ms) / length(latency_ms) : 0))
+    printf("\"min_ms\":%0.3f, \"max_ms\":%0.3f, ", min(latency_ms), max(latency_ms))
+    printf("\"stdev_frames\":%0.6f, \"stdev_ms\":%0.6f }", stddev(latency_frames), stddev(latency_ms))
+    printf("]")
+    fflush()
+  }
+}
+'
+opts_json()
+{
+  local items_=()
+  for idx_ in "${!OPT_NAME[@]}" ; do
+    items_+=("\"${OPT_NAME[$idx_]}\":\"${OPT_VAL[$idx_]}\"")
+  done
+  echo "$(IFS=',' ; printf "%s" "${items_[*]}")"
+}
+
+alsa_device=${OPT_VAL['d']}
+alsa_shorts=$([ "${OPT_VAL['S']}" -eq 1 ] && echo '--shorts' || echo '')
+port_playback=${OPT_VAL['p']}
+port_capture=${OPT_VAL['c']}
+rate=${OPT_VAL['r']}
+run_period=${OPT_VAL['R']}
+run_verbose=$([ "${OPT_VAL['v']}" -eq 1 ] && echo '--verbose' || echo '')
+
+RUN_PERIOD_MAX="$((run_period + 30))s"
+JACKD_TIMEOUT="$((run_period + 15))s"
+JACKD_OPTIONS="${run_verbose} --realtime --temporary"
+JACKD_BACKEND="alsa"
+JACKD_BACKEND_OPTIONS="-d ${alsa_device} -r ${rate} ${alsa_shorts}"
+WAIT_JACKD="2s"
+IODELAY_TIMEOUT="${run_period}s"
+WAIT_IODELAY="2s"
+
+if [ "$port_playback" == "" ] || [ "$port_capture" == "" ];
+then
+  dloge "No playback or capture Jack port is specified. Skip the test."
+  exit 2
+fi
+
+check_jackd_configured()
+{
+  export JACK_NO_AUDIO_RESERVATION=1
+  dlogc "jackd ${JACKD_OPTIONS} -d ${JACKD_BACKEND} ${JACKD_BACKEND_OPTIONS}"
+  # shellcheck disable=SC2086
+  timeout --kill-after ${RUN_PERIOD_MAX} ${run_verbose} ${JACKD_TIMEOUT} \
+      jackd ${JACKD_OPTIONS} -d ${JACKD_BACKEND} ${JACKD_BACKEND_OPTIONS} & jackdPID=$!
+
+  sleep ${WAIT_JACKD}
+
+  dlogc "jack_lsp"
+  jack_lsp -AclLpt
+
+  dlogi "Waiting Jackd to stop without a client"
+  wait ${jackdPID}
+}
+
+collect_latency_data()
+{
+  export JACK_NO_AUDIO_RESERVATION=1
+  dlogc "jackd ${JACKD_OPTIONS} -d ${JACKD_BACKEND} ${JACKD_BACKEND_OPTIONS}"
+  # shellcheck disable=SC2086
+  timeout --kill-after ${RUN_PERIOD_MAX} ${run_verbose} ${JACKD_TIMEOUT} \
+      jackd ${JACKD_OPTIONS} -d ${JACKD_BACKEND} ${JACKD_BACKEND_OPTIONS} & jackdPID=$!
+
+  sleep ${WAIT_JACKD}
+  dlogc "jack_iodelay"
+  # shellcheck disable=SC2086
+  timeout --kill-after ${RUN_PERIOD_MAX} ${run_verbose} ${IODELAY_TIMEOUT} \
+      stdbuf -oL -eL jack_iodelay | tee >(gawk "${AWK_SCRIPT}" > "${LOG_ROOT}/metrics.json") & iodelayPID="$!"
+
+  sleep ${WAIT_IODELAY}
+  dlogi "jack_connect:  ${port_capture} ==>[jack_delay]==> ${port_playback}"
+  jack_connect jack_delay:out "${port_playback}" && jack_connect jack_delay:in "${port_capture}"
+
+  dlogi "Latency data collection"
+  wait ${jackdPID} ${iodelayPID}
+  dlogi "Latency data collection completed."
+}
+
+# TODO: should we set volume to some pre-defined level (parameterized)
+# reset_sof_volume
+# set_alsa_settings
+
+start_test
```

Comment:
https://github.com/thesofproject/sof-test/issues/740 - everything in a function.